clang/lib/Driver/Driver.cpp

```diff
@@ -2443,9 +2441,29 @@ void Driver::BuildInputs(const ToolChain &TC, DerivedArgList &Args,
           if (const char *Ext = strrchr(Value, '.'))
             Ty = TC.LookupTypeForExtension(Ext + 1);
 
+          // For SYCL, convert C-type sources to C++-type sources.
+          if (IsSYCL) {
+            switch (Ty) {
+            case types::TY_C:
+              Ty = types::TY_CXX;
+              break;
+            case types::TY_CHeader:
+              Ty = types::TY_CXXHeader;
+              break;
+            case types::TY_PP_C:
+              Ty = types::TY_PP_CXX;
+              break;
+            case types::TY_PP_CHeader:
+              Ty = types::TY_PP_CXXHeader;
+              break;
+            default:
+              break;
+            }
```

Comment:
Never thought I'd say this as a Python hater, but it's a shame we don't have something analogous to a Python dict :D Take any alternative like a `std::map` - it's too verbose and therefore meaningless for a 4-case switch...

Reply:
In C++23 I hope the pattern matching proposal https://wg21.link/p1371 will be voted in!
Python hater? Do not forget that C++23 is just Python, but running 100x the speed. :-)